plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -77,6 +136,102 @@ protected String getStorageMetadataLocation(String materializedViewName)
+        assertThat(computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isNull();
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        ZonedDateTime lastFreshTime = (ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue();
+        assertThat(lastFreshTime).isNotNull();
+        int result3 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result3).isNotEqualTo(result2);  // mv is not stale anymore so all selects until next refresh returns same result
+        int result4 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        int result5 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result4).isEqualTo(result3);
+        assertThat(result4).isEqualTo(result5);
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isAfter(lastFreshTime);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        int result6 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result6).isNotEqualTo(result5);
+    }
+
+    @Test
+    public void testMaterializedViewCreatedFromPtfAndTable()
+    {
+        String sourceTableName = "source_table_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + sourceTableName + " (VALUE INTEGER)");
+        assertUpdate("INSERT INTO " + sourceTableName + " VALUES(2)", 1);
+        String viewName = "materialized_view_for_ptf_adn_table_" + randomNameSuffix();
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM TABLE(mock.system.sequence_function()) CROSS JOIN " + sourceTableName);
+
+        List<MaterializedRow> materializedRows = computeActual("SELECT * FROM " + viewName + " ORDER BY next_value").getMaterializedRows();
+        assertThat(materializedRows.size()).isEqualTo(1);
+        assertThat(materializedRows.get(0).getField(1)).isEqualTo(2);
+        int valueFromPtf1 = (int) materializedRows.get(0).getField(0);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("STALE");
+        assertThat(computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isNull();
```

Comment:
```suggestion
        assertThat(query("SELECT freshness, last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'"))
                .skippingTypesCheck()
                .matches("VALUES ('STALE', NULL)");
```

Do one query instead of two.

Reply:
IMHO 2 are more readable, also with 2 when it fails you know which condition failed, with 1 you dont